electionleaflets/apps/leaflets/views.py

```diff
@@ -230,23 +240,22 @@ def done(self, form_list, **kwargs):
                     )
                     leaflet.delete()
                     return redirect(reverse("upload_leaflet"))
-                
-                uploaded_images = json.loads(images_text)
-                bucket = self.storage.file_storage.bucket
+
                 for file_path in uploaded_images:
                     file_name = file_path.split("/")[-1]
                     file_name = file_name.replace(" ", "-")
-                    copy_source = {
-                        "Bucket": bucket.name,
-                        "Key": file_path,
-                    }
+                    
+                    storage_backend = getattr(settings, 'DEFAULT_FILE_STORAGE')
+                    bucket = self.storage.file_storage.bucket
```

Comment:
This might fail if the backend doesn't have a `bucket` attribute. Can you move it inside the conditional?

Reply:
Done with https://github.com/DemocracyClub/electionleaflets/pull/259/commits/d9e731a7ecfb63a927ee40f4e2dce14d6d237ad3